### PR TITLE
tests: tfm: secure services: fix bug and add test case for secure read service

### PR DESCRIPTION
**Pull request by andvib**
Verifies only addresses within expected range can be read

### CODE REVIEW

File: tests/modules/tfm/secure_services/src/main.c [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+	zassert_equal(plt_err, TFM_PLATFORM_ERR_SUCCESS, "Valid address returned an error!");
+	zassert_equal(err, 0, "Valid address returned an error!");
+
+	plt_err = tfm_platform_mem_read(output, ficr_end - data_length, data_length, &err);
+	zassert_equal(plt_err, TFM_PLATFORM_ERR_SUCCESS, "Valid address returned an error!");
+	zassert_equal(err, 0, "Valid address returned an error!");
+
+	/* Expect invalid addresses to finish with result -1 */
+	plt_err = tfm_platform_mem_read(output, ficr_start - 1, data_length, &err);
+	zassert_equal(plt_err, TFM_PLATFORM_ERR_SUCCESS,
```
Comment (andvib) [inline tests/modules/tfm/secure_services/src/main.c:64]: @hakonfam I was a bit surprised that the service returns TFM_PLATFORM_ERR_SUCCESS with result set to -1 on invalid addresses. I had a look at the other error values in TFM_PLATFORM and realized there's not that many to choose from. Is returning success even if it was an invalid call accepted in the platform APIs?

Comment (hakonfam) [reply tests/modules/tfm/secure_services/src/main.c]: You are correct, an error should be indicated. I was so rude that I force pushed the fix prior to your commit and then fixed the test to check for the error, please forgive me :)

**Review by hakonfam — APPROVED** [review]

File: tests/modules/tfm/secure_services/src/main.c [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+		return ptr < PM_SRAM_ADDRESS;
+	}
+
+	/* Check if flash address is secure */
+	return ptr < PM_ADDRESS;
+}
+
+void test_tfm_read_service(void)
+{
+	const uint32_t ficr_start = (NRF_FICR_S_BASE + 0x204);
```
Comment (SebastianBoe) [inline tests/modules/tfm/secure_services/src/main.c:28]: On nRF53 NRF_FICR->INFO.CONFIGID is at 0x200.

File: tests/modules/tfm/secure_services/src/main.c [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+	/* Verify that test objects is stored in non-secure addresses. */
+	zassert_true(!is_secure((intptr_t)output),
+			 "Test object is not non-secure");
+
+	/* Verify that secure test objects is stored in secure addresses. */
+	zassert_true(is_secure((intptr_t)secure_ptr),
+						"Test object is not secure");
+
+	/* Verify that the function fails if it ps passed secure pointers */
```
Comment (SebastianBoe) [inline tests/modules/tfm/secure_services/src/main.c:46]: s/ps/is/

**Review by SebastianBoe — CHANGES_REQUESTED** [review]
Changes needed.

File: tests/modules/tfm/secure_services/src/main.c (hunk 2 above)
Comment (andvib) [reply tests/modules/tfm/secure_services/src/main.c:28]: You are correct, I've raised a bug so it can be fixed at a later point

**Review by FrancescoSer — APPROVED** [review]

**Pull request merged**